fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java

```diff
@@ -483,7 +483,9 @@ private double getLengthFromLiteral(LiteralExpr literalExpr, Type type) {
                 }
                 String fileName = Utils.getSuffixName(dirPath, locatedFileStatus.getPath().toString());
                 BlockLocation[] blockLocations = locatedFileStatus.getBlockLocations();
-                List<HdfsFileBlockDesc> fileBlockDescs = getHdfsFileBlockDescs(blockLocations, isObjectStorage(dirPath));
+                List<HdfsFileBlockDesc> fileBlockDescs =
+                        getHdfsFileBlockDescs(blockLocations, isObjectStorage(dirPath)
+                                || (locatedFileStatus.getBlockSize() > Config.hive_max_split_size && isSplittable));
```

Comment:
Put the value of the configuration item in a local variable, cause the configure value can be changed at runtime.

Reply:
nice catch @gengjun-git